### PR TITLE
Spellcheck migration guides

### PR DESCRIPTION
**Pull request by Eric-Arellano**
Closes https://github.com/Qiskit/documentation/issues/442.

### CODE REVIEW

File: cSpell.json [hunk 1]
```diff
@@ -270,10 +270,13 @@
     // words joined by underscores. e.g. hello_world, NEW_KEY_VALUE
     "\\S+_\\S+(_\\S+)*",
     // separate line <span id="" /> tags
-    "^<span id=\\S+ />$"
+    "^<span id=\\S+ />$",
+    "^<span id=\\S+></span>$"
```
Comment (Eric-Arellano) [inline cSpell.json:274]: This matches `span id="foo"></span>` for example.

File: package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "check:metadata": "node -r esbuild-register scripts/commands/checkMetadata.ts",
     "check:links": "node -r esbuild-register scripts/commands/checkLinks.ts",
     "check:release-notes": "node -r esbuild-register scripts/commands/validateReleaseNotesSync.ts",
-    "check:spelling": "cspell --relative --no-progress docs/**/*.md*",
+    "check:spelling": "cspell --relative --no-progress docs/**/*.md* docs/api/**/*.md*",
```
Comment (Eric-Arellano) [inline package.json:12]: Idk why this was necessary...but it was. Without it, the api folder is skipped.

**Comment by Eric-Arellano** [comment]
fyi @abbycross

**Comment by Eric-Arellano** [comment]
> Does this automatically sync the cSpell.json files in inner/open repos or do we need to update them both manually?

No, they need to be synced manually. This fixes it so that we check the migration guides though, which we weren't doing before in open source so it didn't catch the typo you fixed yesterday.

**Comment by abbycross** [comment]
Actually, the spell-check in the inner repo is what caught it! Good to know that we have to update the cSpell in two places when things are changed... can make that part of the process then. FYI @beckykd

**Review by frankharkins — APPROVED** [review]

**Pull request merged**